tools/html_transformer/html_transformer.js

```diff
@@ -0,0 +1,95 @@
+var Promise = require('es6-promise').Promise;
+var jsdom = require('jsdom');
+
+exports.window = null;
+
+exports.load = function(filename) {
+  return new Promise(function(resolve, reject) {
+    jsdom.env(filename, function(errors, window) {
+      if (errors && errors.length) {
+        return reject(errors[0]);
+      }
+
+      exports.window = window;
+      return resolve();
```

Comment:
Yeah that seems fine. In general, I try to only do classes when I know that we'll want to be able to run multiple in the same js context simultaneously since @millermedeiros has convinced me that having your own custom data structures makes it more difficult for others to dive into your codebase